### DISCUSSION TITLE
create stream from USB video graber with seperate audio and video devices

### DISCUSSION THREAD

**Discussion by T-Lancer** [bluenviron/mediamtx#3818] [Questions]
### Question

I am trying to create a stream using a USB video grabber (generic USB 3 HDMI video grabber)

I have video0 as the video source.  and HW:1,0 is the audio stream. The audio stream is unfortunately not linked automatically to the video stream.   I need to mix these together into the RTSP stream, but I am having trouble figuring out the right command.  video works fine, but audio give me an error.

`ffmpeg -re -stream_loop -1 -i /dev/video0 -f alsa  -i hw:1,0  -c:v libx264 -c:a aac copy -f rtsp -rtsp_transport tcp rtsp://192.168.0.2:8557/mystream`

```
[AVFormatContext @ 0x556de76b1780] Unable to choose an output format for 'copy'; use a standard extension for the filename or specify the format manually.
[out#0 @ 0x556de76b1e00] Error initializing the muxer for copy: Invalid argument
Error opening output file copy.
Error opening output files: Invalid argument
```

maybe i ned to use the map command?  but I really am a bit lost here. Maybe someone could provide me with the appropriate command?

thanks!

**Comment by aler9** [comment]
The command you posted is wrong. If you want to encode the audio with the AAC codec, you have to remove "copy":

```
ffmpeg -re -stream_loop -1 -i /dev/video0 -f alsa  -i hw:1,0  -c:v libx264 -c:a aac -f rtsp -rtsp_transport tcp rtsp://192.168.0.2:8557/mystream
```

Otherwise, if you want to keep existing audio encoding, you have to  remove "aac":

```
ffmpeg -re -stream_loop -1 -i /dev/video0 -f alsa  -i hw:1,0  -c:v libx264 -c:a copy -f rtsp -rtsp_transport tcp rtsp://192.168.0.2:8557/mystream
```
